kdl_parser/src/kdl_parser.cpp

```diff
@@ -35,132 +35,138 @@
-  }
-  case urdf::Joint::CONTINUOUS:{
-    Vector axis = toKdl(jnt->axis);
-    return Joint(jnt->name, F_parent_jnt.p, F_parent_jnt.M * axis, Joint::RotAxis);
-  }
-  case urdf::Joint::PRISMATIC:{
-    Vector axis = toKdl(jnt->axis);
-    return Joint(jnt->name, F_parent_jnt.p, F_parent_jnt.M * axis, Joint::TransAxis);
-  }
-  default:{
-    ROS_WARN("Converting unknown joint type of joint '%s' into a fixed joint", jnt->name.c_str());
-    return Joint(jnt->name, Joint::None);
-  }
-  }
-  return Joint();
+  return KDL::Joint();
 }
 
 // construct inertia
-RigidBodyInertia toKdl(urdf::InertialSharedPtr i)
+KDL::RigidBodyInertia toKdl(urdf::InertialSharedPtr i)
 {
-  Frame origin = toKdl(i->origin);
-  
-  // the mass is frame indipendent 
+  KDL::Frame origin = toKdl(i->origin);
+
+  // the mass is frame independent
   double kdl_mass = i->mass;
-  
+
   // kdl and urdf both specify the com position in the reference frame of the link
-  Vector kdl_com = origin.p;
-  
-  // kdl specifies the inertia matrix in the reference frame of the link, 
+  KDL::Vector kdl_com = origin.p;
+
+  // kdl specifies the inertia matrix in the reference frame of the link,
   // while the urdf specifies the inertia matrix in the inertia reference frame
-  RotationalInertia urdf_inertia = 
-    RotationalInertia(i->ixx, i->iyy, i->izz, i->ixy, i->ixz, i->iyz);
-    
+  KDL::RotationalInertia urdf_inertia =
+    KDL::RotationalInertia(i->ixx, i->iyy, i->izz, i->ixy, i->ixz, i->iyz);
+
   // Rotation operators are not defined for rotational inertia,
   // so we use the RigidBodyInertia operators (with com = 0) as a workaround
-  RigidBodyInertia kdl_inertia_wrt_com_workaround =
-    origin.M *RigidBodyInertia(0, Vector::Zero(), urdf_inertia);
-  
+  KDL::RigidBodyInertia kdl_inertia_wrt_com_workaround =
+    origin.M * KDL::RigidBodyInertia(0, KDL::Vector::Zero(), urdf_inertia);
+
   // Note that the RigidBodyInertia constructor takes the 3d inertia wrt the com
   // while the getRotationalInertia method returns the 3d inertia wrt the frame origin
   // (but having com = Vector::Zero() in kdl_inertia_wrt_com_workaround they match)
-  RotationalInertia kdl_inertia_wrt_com = 
+  KDL::RotationalInertia kdl_inertia_wrt_com =
     kdl_inertia_wrt_com_workaround.getRotationalInertia();
-    
-  return RigidBodyInertia(kdl_mass,kdl_com,kdl_inertia_wrt_com);  
+
+  return KDL::RigidBodyInertia(kdl_mass, kdl_com, kdl_inertia_wrt_com);
 }
 
 
 // recursive function to walk through tree
-bool addChildrenToTree(urdf::LinkConstSharedPtr root, Tree& tree)
+bool addChildrenToTree(urdf::LinkConstSharedPtr root, KDL::Tree & tree)
 {
-  std::vector<urdf::LinkSharedPtr > children = root->child_links;
-  ROS_DEBUG("Link %s had %i children", root->name.c_str(), (int)children.size());
+  std::vector<urdf::LinkSharedPtr> children = root->child_links;
+  ROS_DEBUG("Link %s had %i children",
+    root->name.c_str(), static_cast<int>(children.size()));
```

Comment:
could this just be printed as a `size_t` removing the need for an explicit cast?

Reply:
Yeah, I've changed this now.